test/analyzers.test.js

```diff
@@ -0,0 +1,210 @@
+const request = require('request-promise');
+const HttpStatus = require('http-status-codes');
+const WebSocket = require('ws');
+const MongoClient = require('mongodb').MongoClient;
+
+const format = require('util').format;
+
+const video = require('./video/app.js');
+
+const videoAppPort = 8081;
+
+describe('Analyzer Operations: ', () => {
+    describe('green path(create => start => get status => delete): ', () => {
+        let analyzerId = null;
+        let mongoCli = null;
+        let videoApp = null;
+        let testStartTime = (new Date().getTime() / 1000)
+
+        const analyzerInfo = {
+            "name": "Front Gate 1",
+            "source": {
+                "mode": "streaming",
+                "url": "http://localhost:"+videoAppPort+"/video.mp4"
+            },
+            "pipelines": [
+                {
+                    "type": "IntrusionDetection",
+                    "params": {
+                        "roi": [
+                            {"x": 200.1,"y": 100.01},
+                            {"x": 400.1,"y": 100.01},
+                            {"x": 400.1, "y": 600.01},
+                            {"x": 200.1, "y": 600.01}
+                        ],
+                        "triggers": ["person"]
+                    }
+                }
+            ]
+        };
+
+        beforeAll(async () => {
+            videoApp = new video.videoApp(videoAppPort);
+            videoApp.start();
+
+            mongoConn = await MongoClient.connect('mongodb://localhost:27017');
+                const mongoDB = mongoConn.db('jager_test');
+                const analColl = mongoDB.collection('analyzers');
+                const eventColl = mongoDB.collection('events');
+                mongoCli = mongoConn;
+                await analColl.remove({});
+                await eventColl.remove({});
+                mongoConn.close();
+                return;
+        });
+
+        afterAll(() => {
+            videoApp.stop();
+        });
+
+        test('create an analyzer', async (done) => {
+            let postData = analyzerInfo;
+            let options =  {
+                method: 'POST',
+                uri: 'http://localhost:5000/api/v1/analyzers',
+                body: postData,
+                json: true,
+                resolveWithFullResponse: true
+            };
+            let result = await request(options);
+            expect(result.statusCode).toBe(HttpStatus.CREATED);
+            expect(result).toHaveProperty('body');
+            expect(result.body).toHaveProperty('_id');
+            analyzerId = result.body._id;
+            done();
+        });
+        // ----- test('create analyzer')
+
+        test('start the analyzer', async (done) => {
+            expect(analyzerId).not.toBeNull();
```

Comment:
It is wired to test `analyzerId` in this test case because it is already assigned in previous test case. If we want to test the value of `_id`, it's better to test it in **previous test**.